src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +398,241 @@ else if (!period.isEmpty()) {
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_TAG) {
+            // HTML_TAG always has more than 2 children.
+            node = node.getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT
+                // Condition for SINGLETON html element which cannot contain summary node
+                && node.getChildren()[0].getChildren().length > 1) {
+            // Html elements have one tested tag before actual content inside it
+            node = node.getChildren()[0].getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if the first tag inside ast is summary tag.
+     *
+     * @param javadoc root node.
+     * @return true, if first tag is summary tag.
+     */
+    private static boolean isSummaryTag(DetailNode javadoc) {
+        final DetailNode[] child = javadoc.getChildren();
+
+        // Checking size of ast is not required, since ast contains
+        // children of Inline Tag, as at least 2 children will be present which are
+        // RCURLY and LCURLY.
+        return child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                && SUMMARY_TEXT.equals(child[1].getText());
+    }
+
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    private void validateSummaryTag(DetailNode ast) {
+        final String inlineSummary = getInlineSummary(ast);
+        final String summaryVisible = getVisibleContent(inlineSummary);
+        if (summaryVisible.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            if (isPeriodAtEnd(summaryVisible, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
+            }
+            else if (!containsCorrectInlineTags(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Gets whole content of summary tag.
+     *
+     * @param javadoc javadoc root node.
+     * @return Whole content of summary tag.
+     */
+    private String getInlineSummary(DetailNode javadoc) {
+        final FileContents file = getFileContents();
+        final DetailNode[] children = javadoc.getChildren();
+        final int javadocEndLineNo = children[children.length - 1].getLineNumber();
+        final TextBlock textBlock = file.getJavadocBefore(javadocEndLineNo + 1);
+        return Arrays.toString(textBlock.getText());
+    }
+
+    /**
+     * Gets the string that is visible to user in javadoc.
+     *
+     * @param summary Whole content of summary javadoc.
+     * @return string that is visible to user in javadoc.
+     */
+    private static String getVisibleContent(String summary) {
+        final Matcher matcher = SUMMARY_PATTERN.matcher(summary);
```

Comment:
summary should be without summary tag name.

Reply:
"summary" is already without a summary tag name. I am using group so it automatically neglects it.

Reply:
![image](https://user-images.githubusercontent.com/812984/115729750-89303280-a33a-11eb-839c-1b82b5a9d401.png)

inlineSummary variable  should have value "A simple correct Javadoc."

Reply:
Sir, inline summary returns the string we get by `getBlockComment()`. Actual processing is done in `getVisibleContent()` . `inlineSummary` is the raw text we supply to `getVisibleContent` to extract the string that would be visible to user.
![Screenshot from 2021-04-22 19-52-25](https://user-images.githubusercontent.com/69365878/115732934-0cd33900-a3a6-11eb-9c0d-79bafbfb8783.png)

Reply:
Some code changes update:
1. While testing I found that the code used to fail to detect `forbiddenFragments` since it was still multiline. So I had now changed the code such that `getInlineSummary` returns rax text inside summary tag (`@summary` tag excluded) and multiline asterisks removed.
2. `getVisibleSummary` only removes HTML tags and extra white spaces now.

Reply:
after we make https://github.com/checkstyle/checkstyle/issues/9993 code be more logical.
for now lets proceed as is.